src/event-helper-listen.js

```diff
@@ -24,32 +35,85 @@
  * @param {!EventTarget} element
  * @param {string} eventType
  * @param {function(!Event)} listener
- * @param {boolean=} opt_capture
+ * @param {Object=} opt_evtListenerOpts
  * @return {!UnlistenDef}
  */
-export function internalListenImplementation(element, eventType, listener,
-    opt_capture) {
-  let localElement = element;
-  let localListener = listener;
-  /** @type {?Function}  */
-  let wrapped = event => {
-    try {
-      return localListener(event);
-    } catch (e) {
-      // reportError is installed globally per window in the entry point.
-      self.reportError(e);
-      throw e;
-    }
-  };
-  const capture = opt_capture || false;
-  localElement.addEventListener(eventType, wrapped, capture);
-  return () => {
-    if (localElement) {
-      localElement.removeEventListener(eventType, wrapped, capture);
-    }
-    // Ensure these are GC'd
-    localListener = null;
-    localElement = null;
-    wrapped = null;
-  };
-}
+ export function internalListenImplementation(element, eventType, listener,
+   opt_evtListenerOpts) {
+   let localElement = element;
+   let localListener = listener;
+   /** @type {?Function}  */
+   let wrapped = event => {
+     try {
+       return localListener(event);
+     } catch (e) {
+       // reportError is installed globally per window in the entry point.
+       self.reportError(e);
+       throw e;
+     }
+   };
+   const optsSupported = detectEvtListenerOptsSupport();
+   let capture = false;
+   if (opt_evtListenerOpts && opt_evtListenerOpts.capture) {
+     capture = opt_evtListenerOpts.capture;
+   }
+   localElement.addEventListener(
+       eventType,
+       wrapped,
+       optsSupported ? opt_evtListenerOpts : capture
+   );
+   return () => {
+     if (localElement) {
+       localElement.removeEventListener(
+           eventType,
+           wrapped,
+           optsSupported ? opt_evtListenerOpts : capture
+       );
+     }
+     // Ensure these are GC'd
+     localListener = null;
+     localElement = null;
+     wrapped = null;
+   };
+ }
+
+/**
+ * Tests whether the browser supports options as an argument of addEventListener
+ * or not.
+ *
+ * @return {boolean}
+ * @suppress {checkTypes}
+ */
+ export function detectEvtListenerOptsSupport() {
+   if (!self.optsSupported) {
```

Comment:
Setting this on `self` is unnecessary, just use a module level variable. See https://github.com/ampproject/amphtml/blob/master/src/dom.js#L375-L400.